### DISCUSSION TITLE
Возможно ли локальное подключение?

### DISCUSSION THREAD

**Discussion by bazraider** [yandex-maps-unofficial/vue-yandex-maps#272] [Q&A (2.x)] [answered]
Здравствуйте. Интересует вопрос, есть ли возможность подключить vue-yandex-maps v.2.x локально в компоненте Vue, как это было в версии 1.x ?

**Comment by daniluk4000** [comment]
Здравствуйте! Компоненты, несмотря на createYmaps, подключаются локально - эта функция только подключает настройки.

Подскажите, с какой целью вы хотели бы отказаться от регистрации через плагин, и регистрировать строго локально?

**Reply by bazraider** [reply]
Дело в том, что мой проект - это некий монорепозиторий в который входят разные проекты. Таким образом, плагин нужен только в одном из проектов, и для оптимизации не хотелось бы чтобы о плагине знали другие проекты. Я же правильно понимаю, что плагин при такой регистрации попадает в общий бандл?

**Reply by daniluk4000** [reply, marked as answer]
Да, попадет, но не целиком. Так или иначе, в случае с монорепой могу вас понять - даже часть плагина в общем бандле является нарушением подхода монорепы.

Выполните в нужном вам месте функцию `createYmapsOptions`, передав туда нужные настройки. Функция не задокументирована - похоже, стоит всё же добавить возможность красиво менять настройки в API.

Учтите: функцию надо вызывать только один раз, проверок в ней нет. Можно вызывать и бОльшее число раз, но в этом нет смысла и будет небольшая потеря перфоманса.

Предлагаю такой код:

```typescript
import { createYmapsOptions, VueYandexMaps } from 'vue-yandex-maps';

if (!VueYandexMaps.settings.value.apikey) {
    createYmapsOptions({ apikey: '', ... });
}
```

Вызывать функцию надо в created/корне setup, главное не в mounted, в противном случае словите исключение.

А я подумаю, как доработать и задокументировать подход для кейсов по типу ваших на будущее.

**Reply by daniluk4000** [reply]
Создал https://github.com/yandex-maps-unofficial/vue-yandex-maps/issues/273

В целом сам по себе импорт vue-yandex-maps не тянет за собой весь бандл, спасибо нашим сборщикам. Но соглашусь, что импорт из отдельного файла сделает это надежнее

**Reply by bazraider** [reply]
Спасибо большое, попробую предложенный вами вариант

**Reply by daniluk4000** [reply]
Функция улучшена и добавлена в документацию
